main.go

```diff
@@ -121,6 +130,49 @@ func main() {
 					os.Exit(1)
 				}
 			}
+		} else if module, _ := tfconfig.LoadModule(dir); len(module.RequiredCore) >= 1 && len(args) == 0 { //if there is required_version defined in TF code, and no commmand line arguments
+			tfversion := ""
+
+			// we skip duplicated definitions and use only first one
+			tfconstraint := module.RequiredCore[0]
+			tflist, _ := lib.GetTFList(hashiURL, true)
+			fmt.Printf("Reading required version from terraform code, constraint: %s\n", tfconstraint)
+			//fmt.Printf("Available versions: %#v\n", tflist)
+
+			c, err := semver.NewConstraint(tfconstraint)
+			if err != nil {
+				fmt.Println("Error parsing constraint:", err)
+				return
+			}
+			vs := make([]*semver.Version, len(tflist))
+			for i, r := range tflist {
+				v, err := semver.NewVersion(r)
+				if err != nil {
+					fmt.Printf("Error parsing version: %s", err)
+				}
+
+				vs[i] = v
+			}
+
+			sort.Sort(semver.Collection(vs))
+
+			for i := range vs {
+				// Validate a version against a constraint.
+				// Take highest version that matches
+				if c.Check(vs[len(vs)-1-i]) {
+					tfversion = string(vs[len(vs)-1-i].String())
+					fmt.Printf("Matched version: %s\n", tfversion)
+
+					if lib.ValidVersionFormat(tfversion) { //check if version is correct
+						lib.Install(string(tfversion), *custBinPath)
```

Comment:
Shouldn't there be used `break`, or you want intentionally to install multiple versions?